lib/apartment/adapters/sqlite3_adapter.rb

```diff
@@ -13,6 +13,9 @@ def initialize(config)
         @default_dir = File.expand_path(File.dirname(config[:database]))
 
         super
+
+        Apartment.connection_handler = ActiveRecord::ConnectionAdapters::ConnectionHandler.new
+        reset
```

Comment:
Sure :)
I was just suggesting it could be relevant to create the abstraction now since both our PR work around the connection based adapters.